IM_calculation/Advanced_IM/Models/Taghavi_Miranda_2005/run.py

```diff
@@ -0,0 +1,74 @@
+"""
+Python script to run a 3D waveform through Taghavi_Miranda_2005 and store the outputs to a txt file
+"""
+
+import logging
+import os
+
+import numpy as np
+import pandas as pd
+
+from IM_calculation.Advanced_IM import runlibs_2d
+from IM_calculation.IM.Taghavi_Miranda_2005 import Taghavi_Miranda_2005
+from qcore.timeseries import read_ascii
+
+model_dir = os.path.dirname(__file__)
+
+COMPONENT = ["000", "090"]
+ALPHA = [0, 1.5, 5, 15, 30]
+C = [0.02, 0.05, 0.1]
+T = [0.1, 0.4, 0.7, 1.5, 3.0]
+STORIES = 10
+
+
+def main(comp_000, comp_090, output_dir):
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    log_file = os.path.join(output_dir, "log")
+    logging.basicConfig(
+        format="%(asctime)s %(message)s", filename=log_file, level=logging.DEBUG
+    )
+
+    waveforms = {}
+    waveforms["000"], meta = read_ascii(comp_000, meta=True)
+    waveforms["090"] = read_ascii(comp_090)
+    dt = meta["dt"]
+    results = {}
+
+    for component in COMPONENT:
+        results[component] = {}
+        for a in ALPHA:
+            for c in C:
+                for period in T:
+                    im_name = f"TM05_a{a}_c{c}_T{period}_storey"
+                    logging.info(f"calculating {im_name}")
+                    df = Taghavi_Miranda_2005(
+                        waveforms[component], dt, period, a, c, storey=STORIES
+                    )
+                    for i in range(STORIES + 1):
+                        results[component][f"{im_name}{i}_disp_peak"] = df.iloc[
+                            i
+                        ].disp_peak
+                        results[component][f"{im_name}{i}_slope_peak"] = df.iloc[
+                            i
+                        ].slope_peak
+                        results[component][f"{im_name}{i}_storey_shear_peak"] = df.iloc[
+                            i
+                        ].storey_shear_peak
+                        results[component][f"{im_name}{i}_total_accel_peak"] = df.iloc[
+                            i
+                        ].total_accel_peak
+
+    im_csv_fname = os.path.join(output_dir, "Taghavi_Miranda_2005.csv")
+    df = pd.DataFrame.from_dict(results, orient="index")
+    df.index.name = "component"
+    geom = pd.Series(np.sqrt(df.loc["000"] * df.loc["090"]), name="geom")
```

Comment:
maybe use the function to calc geom?

Reply:
the function from runlib? that function includes reading csv (output from opensees) and dumping as well.
the function in runlib has more than one step, and makes more sense to be a function; the case we have here is a simple sqrt command, i'm not sure if creating a new function is a good idea?

Reply:
That makes sense. Probably best to use this one though. So that way we aren't defining geom calculation (even though it's simple) in more than one place.

https://github.com/ucgmsim/IM_calculation/blob/cb29fc5faa69ff95444a255a2b75c18e7186656e/IM_calculation/IM/intensity_measures.py#L147

Reply:
Yeah, use the get_geom function.

One day we might change geom to be exp((log(d1)+log(d2))/2)

Reply:
the latest commit should have updated the code to use get_geom from IM_calculation.IM.intensity_measures

Reply:
@jasonmotha since @jpa198 mentioned this potential change, should the geom calculation in OpenSees's 2D model be updated to have this as well(using get_geom)? or should they stay independent?

Reply:
Yes. 

```        line = np.sqrt(ims.loc["090"] * ims.loc["000"])````
        
        should be updated to use the function.

Reply:
Please note: The alternate formula is mathematically correct, but is dead slow compared to the current sqrt method.
All geom calculations should still point at the one function though